### ISSUE TITLE
E: shared-lib-without-dependency-information /usr/bin/xxx for GO binaries

### ISSUE THREAD

**Issue by socketpair** [rpm-software-management/rpmlint#598]
I use Fedora 33 and rpmlint from the distro (rpmlint-1.11-12.fc33.noarch)

Fragment  of .spec file how I build my GO sources:
```sh
%build
export GOPATH=/usr/share/gocode
export GOBUILDFLAGS="%gobuildflags"
export CGO_ENABLED=0
mkdir bin
export GOBIN=$(pwd)/bin
eval "go install $GOBUILDFLAGS ./mysources/..."
```

Example of the binary:
[bug.zip](https://github.com/rpm-software-management/rpmlint/files/5794106/bug.zip)

**Comment by marxin** [comment]
Thank you for the report.

Right now we skip `.gox` files
https://github.com/rpm-software-management/rpmlint/blob/7a87b6c870fdde6681d3a3a1f19555b26ddf2981/rpmlint/checks/BinariesCheck.py#L614-L618

shouldn't the installed `bug` be named with the extension? I'm sorry, but I'm not much familiar with `go` extensions, but I know that we intentionally skip ELFs created with go for various reasons.

**Comment by socketpair** [comment]
No! my binaries are not extensions! they are final executable files. That's why they are placed in /usr/bin

**Comment by marxin** [comment]
All right, we can filter out the checks based on:

```
$ readelf -n bug
...
Displaying notes found in: .note.go.buildid
  Owner                Data size 	Description
  Go                   0x00000053	Unknown note type: (0x00000004)
   description data: 32 35 37 38 74 7a 4d 61 66 2d 68 32 39 62 37 4a 36 5f 47 46 2f 43 49 39 43 36 71 64 62 70 65 5a 79 79 5a 35 6e 39 4f 36 48 2f 6a 78 7a 65 6f 30 61 50 76 6c 4f 74 51 45 4d 55 71 6b 6e 31 2f 37 64 50 39 6d 64 61 31 6d 50 41 4a 43 6e 2d 6d 49 66 35 42 
```

I'll work on that..

**Comment by marxin** [comment]
@socketpair May I please ask you for steps on how to create a reproducer that will be as small as possible?

**Comment by socketpair** [comment]
What should I give to you ? minimized srpm ?

**Comment by marxin** [comment]
> @marxin What should I give to you ? minimized srpm ?

Yes, please.

**Comment by socketpair** [comment]
[example.zip](https://github.com/rpm-software-management/rpmlint/files/5796286/example.zip)

srpm + generated rpms

**Comment by marxin** [comment]
@Conan-Kudo Ok, looking again at the provided `/usr/bin/example1`, I see the following:

```
$ file ./usr/bin/example1
./usr/bin/example1: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, BuildID[sha1]=8b6123ebd92047ed4d7edefd646d453b98332dba, stripped
```

So it's classified as "shared object" and it's really statically linked:
```
ldd ./usr/bin/example1
	statically linked
```

That said, the error message looks reasonable to me.

**Comment by socketpair** [comment]
@marxin no such commit in master. Please reopen.

**Comment by marxin** [comment]
Sorry, you're correct as #599 was not merged.
